go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -294,3 +294,43 @@ func runOvnKube(ctx *cli.Context) error {
 	wg.Wait()
 	return nil
 }
+
+// wait for OVNNB server to be up and running
+func waitForOVNNBServer(ctx *cli.Context) (goovn.Client, error) {
+	timeout := time.After(10 * time.Second)
```

Comment:
@msherif1234 I was going to suggest something from `k8s.io/apimachinery/pkg/util/wait` but none of those are quite right (eg we'd want PollImmediateUntilWithContext). But both these functions are doing almost the same thing and should be combined. Instead, you'd:

```
		if ovnSBClient, err = waitForOVNDBServer(ctx, util.NewOVNNBClient); err != nil {
			return fmt.Errorf("error when trying to initialize go-ovn SB client: %v", err)
		}

func waitForOVNDBServer(ctx context.Context, createFn func() (goovn.Client, err)) (goovn.Client, error) {
	timeout := time.After(10 * time.Second)
	var client goovn.Client
	for {
		select {
		case <-ctx.Done():
			return client, fmt.Errorf("canceled waiting for OVNNB server")
		case <-timeout:
			return client, fmt.Errorf("timed out waiting for OVNNB server")
		default:
			if client, err := createFn(); err == nil {
				return client, nil
			}
			// try again later
			time.Sleep(100 * time.Millisecond)
		}
	}
}
```

Reply:
sure